middleware/common/include/common/algorithm/compare.h

```diff
@@ -18,6 +18,12 @@ namespace casual
          return ( equal( value, values) || ... ); 
       }
 
+      template< typename V, typename... Vs>
+      constexpr bool none( V&& value, Vs&&... values)
+      {
+         auto not_equal = []( auto&& l, auto& r){ return l != r;};
+         return ( not_equal( value, values) && ... );
```

Comment:
While we're at it, we should remove the hack for the g++ 9 bug. Just compare directly in the fold expression.
`return ( ( value != values) && ...);`

Also for the `none( ...)` above.

Reply:
I have verified that it is fixed on g++ 13.2 (probably a lot earlier)